### DISCUSSION TITLE
Batch cell updates

### DISCUSSION THREAD

**Discussion by doczoidberg** [handsontable/hyperformula#553] [General]
1. When are the computations of a cell and it's dependencies are executed? By setting the cell content or by getting the value of a cell?

2. Is it possible to batch cell updates. When updating multiple cells it would be faster when one can lock the computation, then setting the cells and then unlocking the computation.

**Comment by wojciechczerniak** [comment]
@doczoidberg Have you seen https://handsontable.github.io/hyperformula/guide/batch-operations.html#how-to-batch ?

Answering your questions. 
1. Computations are executed on get, but dependencies are resolved on set. That's why we have `batch` API.
2. It is possible to batch cell updates, see the link above.

**Comment by doczoidberg** [comment]
I think `suspendEvaluation` and `resumeEvaluation` do not work correct with cross references? Seems that references on other sheets are not updated after `resumeEvaluation`.

**Comment by wojciechczerniak** [comment]
@doczoidberg Can you provide step by step to reproduce the problem? Just to be sure we're on the same page. I will ping @bardek8 @izulin and @voodoo11 to check it.

For example:

1. `suspendEvaluation`
2. Add sheet or cell with formula that has a reference to other sheet? (existent or not yet? can you add example formula?)
3. `resumeEvaltuation`

**Comment by doczoidberg** [comment]
what I do.
1. add different sheets (sheeta, sheetb...)
2. `suspendEvaluation`
3. multiple `hf.setCellContents` of sheeta
4. `resumeEvaluation`
5. cells in sheetb seem not to be updated, I get different results without suspending/resuming

**Comment by izulin** [comment]
> 1. Computations are executed on get, but dependencies are resolved on set. That's why we have `batch` API.

Clarification: both computations and dependencies are resolved on set.

**Comment by doczoidberg** [comment]
this was my fault. I tried to read cells in a batch run.
